test/core/storage/arango_sample_storage_scrubber_test.py

```diff
@@ -0,0 +1,182 @@
+    db.create_collection(TEST_COL_SCHEMA)
+    return db
+
+
+def samplestorage_method(arango):
+    clear_db_and_recreate(arango)
+    return ArangoSampleStorage(
+        arango.client.db(TEST_DB_NAME, TEST_USER, TEST_PWD),
+        TEST_COL_SAMPLE,
+        TEST_COL_VERSION,
+        TEST_COL_VER_EDGE,
+        TEST_COL_NODES,
+        TEST_COL_NODE_EDGE,
+        TEST_COL_WS_OBJ_VER,
+        TEST_COL_DATA_LINK,
+        TEST_COL_SCHEMA)
+
+def dt(timestamp):
+    return datetime.datetime.fromtimestamp(timestamp, tz=datetime.timezone.utc)
+
+def _create_and_expire_data_link(samplestorage, link, expired, user):
+    samplestorage.create_data_link(link)
+    samplestorage.expire_data_link(expired, user, link.id)
+
+def test_timestamp_seconds_to_milliseconds(samplestorage):
+    ts1=1614958000000 # milliseconds
+    ts2=1614958000    # seconds
+    ts3=1614958       # seconds
+    ts4=9007199254740.991 # seconds
+
+    id1 = uuid.UUID('1234567890abcdef1234567890abcdef')
+    id2 = uuid.UUID('1234567890abcdef1234567890abcdee')
+    assert samplestorage.save_sample(
+        SavedSample(id1, UserID('user'), [SampleNode('mynode')], dt(ts3), 'foo')) is True
+    assert samplestorage.save_sample_version(
+        SavedSample(id1, UserID('user'), [SampleNode('mynode1')], dt(ts2), 'foo')) == 2
+    assert samplestorage.save_sample(
+        SavedSample(id2, UserID('user'), [SampleNode('mynode2')], dt(ts3), 'foo')) is True
+
+    lid1=uuid.UUID('1234567890abcdef1234567890abcde2')
+    lid2=uuid.UUID('1234567890abcdef1234567890abcde3')
+    lid3=uuid.UUID('1234567890abcdef1234567890abcde4')
+    samplestorage.create_data_link(DataLink(
+        lid1,
+        DataUnitID(UPA('42/42/42'), 'dataunit1'),
+        SampleNodeAddress(SampleAddress(id1, 1), 'mynode'),
+        dt(ts2),
+        UserID('user'))
+    )
+
+    samplestorage.create_data_link(DataLink(
+        lid2,
+        DataUnitID(UPA('5/89/32'), 'dataunit2'),
+        SampleNodeAddress(SampleAddress(id2, 1), 'mynode2'),
+        dt(ts3),
+        UserID('user'))
+    )
+
+    _create_and_expire_data_link(
+        samplestorage,
+        DataLink(
+            lid3,
+            DataUnitID(UPA('5/89/33'), 'dataunit1'),
+            SampleNodeAddress(SampleAddress(id1, 1), 'mynode'),
+            dt(ts3),
+            UserID('user')),
+        dt(ts3+100),
+        UserID('user')
+    )
+
+    assert samplestorage.get_sample(id1, 1).savetime == dt(ts3)
+    assert samplestorage.get_sample(id1, 2).savetime == dt(ts2)
+    assert samplestorage.get_sample(id2).savetime == dt(ts3)
+    assert samplestorage.get_data_link(lid1).created == dt(ts2)
+    assert samplestorage.get_data_link(lid2).created == dt(ts3)
+    assert samplestorage.get_data_link(lid3).created == dt(ts3)
+    assert samplestorage.get_data_link(lid3).expired == dt(ts3+100)
+
+    samplestorage._db.aql.execute(
+        """
+        FOR sample1 IN samples_nodes
+            FILTER sample1.saved < 1000000000000
```

Comment:
Let's put this make this lower threshold a parameter and maybe explain the rationale.